wandb/sdk/wandb_run.py

```diff
@@ -2587,10 +2580,6 @@ def _console_stop(self) -> None:
             self._output_writer = None
 
     def _on_start(self) -> None:
-        # would like to move _set_global to _on_ready to unify _on_start and _on_attach
-        # (we want to do the set globals after attach)
-        # TODO(console) However _console_start calls Redirect that uses `wandb.run` hence breaks
-        self._set_globals()
```

Comment:
is the console comment still an issue? if not i guess we can move it to on_start and we have access to the settings so it can be skip for "create_new"

Reply:
I moved the responsibility of calling `_set_globals()` out to `wandb_init.py`, and in a different PR I updated `redirect.py` to not use `wandb.run`.

Reply:
yeah i saw, you believe it is better instead of unifying it in wandb.Run?

Reply:
Yeah I think so. I don't think the run itself should be aware of how references to it are stored.